plugins/woocommerce/src/Blocks/Templates/SingleProductTemplate.php

```diff
@@ -57,7 +57,24 @@ public function render_block_template() {
 			}
 			$templates = get_block_templates( array( 'slug__in' => $valid_slugs ) );
 
-			if ( isset( $templates[0] ) && BlockTemplateUtils::template_has_legacy_template_block( $templates[0] ) ) {
+			if ( count( $templates ) === 0 ) {
+				return;
+			}
+
+			// Use the first template by default.
+			$template = $templates[0];
+
+			// Check if there is a template matching the slug `single-product-{post_name}`.
+			if ( count( $valid_slugs ) > 1 && count( $templates ) > 1 ) {
+				foreach ( $templates as $t ) {
+					if ( $valid_slugs[1] === $t->slug ) {
```

Comment:
This code feels a bit clunky, assuming we'd never add another valid slug rather than just checking if the slug matches the one assigned in a previous conditional.

It may just be my preference but I think i would refactor this a little bit so that we're not directly referencing what feels like arbitrary array indexes that could change. I think I'd change the code on line 54/55/56:

```php
$valid_slugs = array( self::SLUG );
$single_product_slug = 'product' === $post->post_type && $post->post_name  ? 'single-product-' . $post->post_name : '';

if ($single_product_slug) {
  $valid_slugs[] = $single_product_slug;
}

// then line 68
if ( $single_product_slug ) {
   foreach ( $templates as $t ) {
		if ( $single_product_slug === $t->slug ) {
```

Reply:
I like your suggestion. I updated the logic in d06a7aa8b6e1af2866f0453dc7d19dd192d784ee. Let me know how it looks.

Reply:
Looks perfect 👌🏻